Libraries/libbom/Headers/bom/bom.h

```diff
@@ -42,6 +42,9 @@ struct bom_context *
 bom_alloc_empty(struct bom_context_memory memory);
 
 struct bom_context *
+bom_alloc_size(struct bom_context_memory memory, uint32_t index_count);
```

Comment:
I'd just merge this extra parameter into `bom_alloc_empty()`, easy enough to just pass zero if you don't know the size.